Microsoft.O365.Security.Native.ETW/EventRecordMetadata.hpp

```diff
@@ -153,6 +153,68 @@ namespace Microsoft { namespace O365 { namespace Security { namespace ETW {
             return dest;
         }
 
+#pragma endregion
+
+#pragma region ExtendedData
+
+        /// <summary>
+        /// If the event's extended data contains an Argon container ID, retrieve it.
+        /// Can be expensive, avoid calling more than once per event.
+        /// </summary>
+        /// <returns>
+        /// A Guid representing the Argon container ID if the data is present. Null it's not present. 
+        /// Throws a ContainerIdFormatException if the container ID is present but parsing fails.
+        /// </returns>
+        virtual System::Nullable<System::Guid> GetContainerId()
```

Comment:
I'd prefer to avoid introducing `nullable` here since it puts extra work on each caller. could we instead define a public No_Container_Id as Guid.Empty (00000000-0000-0000-0000-0000000000000) and return that when no container ID is present?

Alternatively we could do `bool TryGetContainerId(out Guid ContainerId)` here which is idiomatic for cases like this.

Reply:
Went with the TryGetContainerId suggestion - I feel like it's useful to draw a distinction between a valid but possibly erroneous Guid value and a missing value.